src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -1059,6 +1059,37 @@ CDRMessage_t PDP::get_participant_proxy_data_serialized(
     return cdr_msg;
 }
 
+ParticipantProxyData* PDP::get_participant_proxy_data(
+        const GuidPrefix_t& guid_prefix)
+{
+    for (auto pit = ParticipantProxiesBegin(); pit != ParticipantProxiesEnd(); ++pit)
+    {
+        if (guid_prefix == (*pit)->m_guid.guidPrefix)
+        {
+            return *(pit);
+        }
+    }
+    return nullptr;
+}
+
+bool PDP::is_known_participant(
+        const GuidPrefix_t& guid_prefix)
+{
+    for (auto pit = ParticipantProxiesBegin(); pit != ParticipantProxiesEnd(); ++pit)
+    {
+        if (guid_prefix == (*pit)->m_guid.guidPrefix)
+        {
+            return true;
+        }
+    }
+    return false;
```

Comment:
nit: it reduces duplication code.
```suggestion
    return (nullptr != get_participant_proxy_data(guid_prefix));
```

Reply:
Done in 8b6a662